src/GitHub.VisualStudio/TeamExplorer/Connect/GitHubConnectSection.cs

```diff
@@ -129,23 +183,77 @@ void UpdateConnection()
         void OnPropertyChange(object sender, System.ComponentModel.PropertyChangedEventArgs e)
         {
             if (e.PropertyName == "IsVisible" && IsVisible && View == null)
-                View = new GitHubConnectContent() { DataContext = this };
+                View = new GitHubConnectContent { DataContext = this };
+        }
+
+        void UpdateIcons(object sender, NotifyCollectionChangedEventArgs e)
+        {
+            if (e.Action == NotifyCollectionChangedAction.Add)
+            {
+                e.NewItems
+                    .Cast<ISimpleRepositoryModel>()
+                    .Select(async r =>
+                {
+                    if (String.Equals(Holder.ActiveRepo?.RepositoryPath, r.LocalPath, StringComparison.CurrentCultureIgnoreCase))
+                        SelectedRepository = r;
+                    var repo = await ApiFactory.Create(r.CloneUrl).GetRepository();
+                    r.SetIcon(repo.Private, repo.Fork);
+                }).ToList();
+            }
         }
 
+        void UpdateRepositoryList(object sender, PropertyChangedEventArgs e)
+        {
+            if ((isCloning || isCreating) && e.PropertyName == "IsBusy" && !((ITeamExplorerSection)sender).IsBusy)
+            {
+                var vsservices = ServiceProvider.GetExportedValue<IVSServices>();
+                if (isCreating)
+                {
+                    vsservices.ClearNotifications();
+                    vsservices.ShowMessage(string.Format(CultureInfo.CurrentUICulture, "[{0}]({1} has been successfully created.", SelectedRepository.Name, SelectedRepository.CloneUrl));
```

Comment:
In the future, you can target one PR towards another PR's branch. That way only this one commit would show up. :smile: